js/nextperson.js

```diff
@@ -0,0 +1,75 @@
+(() => {
+    let index = 0;
+
+    const peronName = document.getElementById('next-people__name');
+    const peronHeight = document.getElementById('next-people__height');
+    const peronHairColor = document.getElementById('next-people__hair');
+    const peronSkinColor = document.getElementById('next-people__skin');
+    const peronBirthYear = document.getElementById('next-people__birth');
+    const peronGender = document.getElementById('next-people__gender');
+
+    const nextPersonBtn = document.getElementById('next');
+
+    const swapiPeopleAPI1 = (index = 1) => `https://swapi.co/api/people/${index}/?format=json`;
+    const swapiPeopleAPI2 = (index = 1) => `https://swapi.co/api/people/${index}/?format=json`;
+    const swapiPeopleAPI3 = (index = 1) => `https://swapi.co/api/people/${index}/?format=json`;
+    const swapiPeopleAPI4 = (index = 1) => `https://swapi.co/api/people/${index}/?format=json`;
+    const swapiPeopleAPI5 = (index = 1) => `https://swapi.co/api/people/${index}/?format=json`;
+    const swapiPeopleAPI6 = (index = 1) => `https://swapi.co/api/people/${index}/?format=json`;
+    console.log('promise executed successfully1');
+
+    function getPerson1(index = 1) {
+        return fetch(swapiPeopleAPI1(index)).then(data => data.json());
+
+    }
+
+    function getPerson2(index = 1) {
+        return fetch(swapiPeopleAPI2(index)).then(data => data.json());
+
+    }
+
+    function getPerson3(index = 1) {
+        return fetch(swapiPeopleAPI3(index)).then(data => data.json());
+
+    }
+
+    function getPerson4(index = 1) {
+        return fetch(swapiPeopleAPI4(index)).then(data => data.json());
+
+    }
+
+    function getPerson5(index = 1) {
+        return fetch(swapiPeopleAPI5(index)).then(data => data.json());
+
+    }
+
+    function getPerson6(index = 1) {
+        return fetch(swapiPeopleAPI6(index)).then(data => data.json());
+
+    }
+    console.log('promise executed successfully2');
+    console.log(getPerson1())
+    async function updateNextPerson() {
+        const { name } = await getPerson1(++index);
```

Comment:
again use one function to get details and have all properties destructured on the same like

Reply:
you can also make the same function to take category make it reusable for categories